lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -509,10 +511,27 @@ void ExtractInstancesPass::extractInstances() {
         oldParentInst.getResult(portIdx).replaceAllUsesWith(
             newParentInst.getResult(portIdx));
 
+      // Ensure that the `inner_sym` of the instance is unique within the parent
+      // module we're extracting it to.
+      StringAttr newInstSym{};
+      if (auto oldInstSym = inst.inner_symAttr()) {
+        auto uniquifiedName =
+            getModuleNamespace(newParent).newName(oldInstSym.getValue());
+        if (uniquifiedName != oldInstSym.getValue()) {
+          newInstSym = StringAttr::get(&getContext(), uniquifiedName);
+          LLVM_DEBUG(llvm::dbgs() << "  - Uniquified symbol " << oldInstSym
+                                  << " to " << newInstSym << "\n");
+        } else {
+          newInstSym = oldInstSym;
+        }
+      }
```

Comment:
I would move this code after creating the instance op.

```suggestion
      // Ensure that the `inner_sym` of the instance is unique within the parent
      // module we're extracting it to.
      if (auto instSym = inst.inner_symAttr();) {
        auto newName = getModuleNamespace(newParent).newName(oldInstSym.getValue());
        if (newName != instSym.getValue())
          inst.inst_symAttr(StringAttr::get(&getContext(), newName));
      }
```

Reply:
Yeah good suggestion! I'll just have to make sure to do this before the instance gets inserted into its new parent, otherwise `getModuleNamespace` would make its symbol collide with itself.